GANDLF/cli/huggingface_hub_handler.py

```diff
@@ -0,0 +1,52 @@
+from huggingface_hub import HfApi, snapshot_download
+from typing import List, Union
+
+
+def push_to_model_hub(
+    repo_id: str,
+    folder_path: str,
+    path_in_repo: Union[str, None] = None,
+    commit_message: Union[str, None] = None,
+    commit_description: Union[str, None] = None,
+    token: Union[str, None] = None,
+    repo_type: Union[str, None] = None,
+    revision: Union[str, None] = None,
+    allow_patterns: Union[List[str], str, None] = None,
+    ignore_patterns: Union[List[str], str, None] = None,
+    delete_patterns: Union[List[str], str, None] = None,
+):
+    api = HfApi(token=token)
+
+    api.create_repo(repo_id, exist_ok=True)
+
+    api.upload_folder(
+        repo_id=repo_id,
+        token=token,
```

Comment:
```suggestion
    api = HfApi(token=token)

    repo_url = api.create_repo(repo_id, exist_ok=True)
    repo_id = repo_url.repo_id

    api.upload_folder(
        repo_id=repo_id,
```

(nit) 2 things:
- `token` is already passed to `HfApi` so no need to repeat it in `upload_folder`
- `repo_id` provided by the user might be implicit (e.g. `"my-cool-repo"`) which `create_repo` can handle. The returned value gives you the actual repo id (e.g. `"Wauplin/my-cool-repo"`). All other methods, including `upload_folder`, needs the explicit repo id so better to retrieve it. TL;DR: user will be able to do `push_to_hub("my-cool-model", ...)`.

Reply:
@Wauplin, You're right. Commiting your suggestion.